client/App.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { HashRouter as Router, Link, Route } from 'react-router-dom'
+import NavBar from './components/NavBar'
+import Footer from './components/Footer'
+import Categories from './components/Categories'
+import CategoryProducts from './components/CategoryProducts'
+import SingleProduct from './components/SingleProduct'
+
+
+class App extends React.Component {
+    constructor() {
+        super()
+    };
+
+    render() {
+        return (
+            <Router>
+                <div id='home'>
+                    <NavBar />
+                    <Route path='/' exact component={ Categories } />
```

Comment:
instead of mixing up all the authorized routes with the unauthorized routes, you can use a Switch (which kinda works just like a switch statement)

example:
```suggestion
 import { Switch } from 'react-router-dom';
<Switch>
        {/* Routes placed here are available to all visitors */}
        {isLoggedIn && (
          <Switch>
            {/* Routes placed here are only available after logging in */}
          </Switch>
        )}
        {/* Displays our Login component as a fallback */}
</Switch>
```